### PR TITLE
Travis CI improvements

### PR DESCRIPTION
**Pull request by atodorov**
First fix all flake8 errors and warnings. Then I want to refactor the Makefile to not return an error about missing rpm command in Travis.

### CODE REVIEW

**Comment by atodorov** [comment]
@tkdchen - I think you should review & merge this one before continuing with the rest of my PRs. I'll see if I can add `make test` to Travis as well.

**Comment by tkdchen** [comment]
This is awesome. Thank you very much.

BTW, is it possible to select a RPM based system in travis-ci?

**Comment by atodorov** [comment]
> BTW, is it possible to select a RPM based system in travis-ci?

NO, as far as I know, they are Ubuntu/Debian based. If that is possible you should check their documentation or ask their support channel.

File: tcms/xmlrpc/__init__.py [hunk 1]
```diff
@@ -17,7 +20,3 @@
 
 def get_version():
     return XMLRPC_VERSION
-
-
-from tcms.xmlrpc.filters import autowrap_xmlrpc_apis
-autowrap_xmlrpc_apis(__path__, __package__)
```
Comment (tkdchen) [inline tcms/xmlrpc/__init__.py]: I just found this change causes following error

```
(nitrate)[cqi@cqi-laptop Nitrate]$ ./manage.py runserver --traceback
Traceback (most recent call last):
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/core/management/base.py", line 222, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/core/management/base.py", line 250, in execute
    translation.activate('en-us')
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/utils/translation/__init__.py", line 90, in activate
    return _trans.activate(language)
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/utils/translation/trans_real.py", line 183, in activate
    _active.value = translation(language)
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/utils/translation/trans_real.py", line 172, in translation
    default_translation = _fetch(settings.LANGUAGE_CODE)
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/utils/translation/trans_real.py", line 154, in _fetch
    app = import_module(appname)
  File "/home/cqi/Virtualenvs/nitrate/lib/python2.7/site-packages/django/utils/importlib.py", line 35, in import_module
    __import__(name)
  File "/home/cqi/code/Nitrate/tcms/xmlrpc/__init__.py", line 16, in <module>
    autowrap_xmlrpc_apis(__path__, __package__)
  File "/home/cqi/code/Nitrate/tcms/xmlrpc/filters.py", line 85, in autowrap_xmlrpc_apis
    _wrap_exceptions(module_name)
  File "/home/cqi/code/Nitrate/tcms/xmlrpc/filters.py", line 36, in _wrap_exceptions
    module = __import__(module_name, {}, {}, [""])
  File "/home/cqi/code/Nitrate/tcms/xmlrpc/api/__init__.py", line 13, in <module>
    from version import *  # NOQA
  File "/home/cqi/code/Nitrate/tcms/xmlrpc/api/version.py", line 4, in <module>
    from tcms.xmlrpc import get_version
ImportError: cannot import name get_version
```

I'll make a commit to fix this right away.

**Pull request merged**